.github/workflows/testing.yml

```diff
@@ -240,3 +240,191 @@ jobs:
+          #
+          # In brief: when "src/lib/small" is in include paths,
+          # `#include <version>` from inside Mac OS SDK headers
+          # attempts to include "src/lib/small/VERSION" as a
+          # header file that leads to a syntax error.
+          #
+          # It was fixed in the following commits:
+          #
+          # * 1.10.10-24-g7bce4abd1
+          # * 2.7.2-44-gbb1d32903
+          # * 2.8.1-56-ga6c29c5af
+          # * 2.9.0-84-gc5ae543f3
+          #
+          # However applying the workaround for all versions looks
+          # harmless.
+          #
+          # Added -f just in case: I guess we'll drop this useless
+          # obsoleted VERSION file from the git repository sooner
+          # or later.
+          rm -f src/lib/small/VERSION
+          # The same as above, but for the VERSION file generated
+          # by tarantool's CMake script.
+          rm VERSION
+          # }}} Workaround Mac OS build failure (gh-6076)
+          # Continue the build.
+          make -j$(sysctl -n hw.logicalcpu)
+          make install
+        if: matrix.tarantool != 'brew' && steps.cache.outputs.cache-hit != 'true'
+
+      - name: Install tarantool
+        run: |
+          cd "${T_TARDIR}"
+          make install
+        if: matrix.tarantool != 'brew' && steps.cache.outputs.cache-hit == 'true'
+
+      - name: Verify tarantool version
+        run: |
+          # Workaround https://github.com/tarantool/tarantool/issues/4983
+          # Workaround https://github.com/tarantool/tarantool/issues/5040
+          tarantool -e "require('fiber').sleep(0) assert(_TARANTOOL:startswith('${T_VERSION}'), _TARANTOOL) os.exit()"
+        if: matrix.tarantool != 'brew' && matrix.tarantool != 'master'
+
+      - name: Setup golang for the connector and tests
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.golang }}
+
+      - name: Install test dependencies
+        run:  |
+          brew install luarocks
+          cd "${SRCDIR}"
+          make deps
+
+      - name: Run regression tests
+        run: |
+          cd "${SRCDIR}"
+          make test
+
+      - name: Run regression tests with call_17
+        run: |
+          cd "${SRCDIR}"
+          make test TAGS="go_tarantool_call_17"
+
+      - name: Run regression tests with msgpack.v5
+        run: |
+          cd "${SRCDIR}"
+          make test TAGS="go_tarantool_msgpack_v5"
+
+      - name: Run regression tests with msgpack.v5 and call_17
+        run: |
+          cd "${SRCDIR}"
+          make test TAGS="go_tarantool_msgpack_v5,go_tarantool_call_17"
+
+      - name: Run fuzzing tests
+        if: ${{ matrix.fuzzing }}
+        run: |
+          cd "${SRCDIR}"
+          make fuzzing TAGS="go_tarantool_decimal_fuzzing"
+
+      - name: Check workability of benchmark tests
+        run: |
+          cd "${SRCDIR}"
+          make bench-deps bench DURATION=1x COUNT=1
```

Comment:
Could we extract tests in a separate GitHub action/function instead of copying it?

Reply:
I don't think so, because we make tests differently for linux/macOs.
ee:
```
      - name: Run regression tests with call_17
        run: |
          source tarantool-enterprise/env.sh
          make test TAGS="go_tarantool_call_17"
          make testrace TAGS="go_tarantool_call_17"
        env:
          TEST_TNT_SSL: ${{matrix.ssl}}
```

ce:
```
      - name: Run regression tests with call_17
        run: |
          make test TAGS="go_tarantool_call_17"
```

ee:
```
      - name: Run regression tests with call_17
        run: |
          cd "${SRCDIR}"
          make test TAGS="go_tarantool_call_17"
```

Reply:
Why are there no `testrace` on macOS?

Reply:
added